File: Leetcode/Python/2965. Find Missing and Repeated Values/solution.py
# https://leetcode.com/problems/find-missing-and-repeated-values/description

class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        exist = [False] * ((len(grid) ** 2) + 1)
        result = [0, 0]

        for y in range(len(grid)):
            for x in range(len(grid[0])):
                if exist[grid[y][x]]:
                    result[0] = grid[y][x]

                exist[grid[y][x]] = True

        for idx in range(1, len(exist)):
            if not exist[idx]:
                result[1] = idx
                return result

        return [-1, -1]

File: Leetcode/Typescript/2965. Find Missing and Repeated Values/solution.ts
// https://leetcode.com/problems/find-missing-and-repeated-values/description

function findMissingAndRepeatedValues(grid: number[][]): number[] {
  const count = Array.from({ length: grid.length ** 2 + 1 }, (_) => 0);

  for (let y = 0; y < grid.length; y++) {
    for (let x = 0; x < grid.length; x++) {
      count[grid[y][x]] += 1;
    }
  }

  const result = [0, 0];

  for (let num = 1; num < count.length; num++) {
    if (count[num] === 0) {
      result[1] = num;
    } else if (count[num] === 2) {
      result[0] = num;
    }
  }

  return result;
}
